spring/spring4/spring4-remote/src/main/java/org/infinispan/spring/session/RemoteApplicationPublishedBridge.java

```diff
@@ -0,0 +1,45 @@
+package org.infinispan.spring.session;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.annotation.ClientCacheEntryCreated;
+import org.infinispan.client.hotrod.annotation.ClientCacheEntryExpired;
+import org.infinispan.client.hotrod.annotation.ClientCacheEntryRemoved;
+import org.infinispan.client.hotrod.annotation.ClientListener;
+import org.infinispan.client.hotrod.event.ClientCacheEntryCreatedEvent;
+import org.infinispan.client.hotrod.event.ClientCacheEntryExpiredEvent;
+import org.infinispan.client.hotrod.event.ClientCacheEntryRemovedEvent;
+import org.infinispan.spring.provider.SpringCache;
+
+/**
+ * A bridge between Infinispan Remote events and Spring.
+ *
+ * @author Sebastian Łaskawiec
+ * @since 9.0
+ */
+@ClientListener
+public class RemoteApplicationPublishedBridge extends AbstractApplicationPublisherBridge {
+
+   public RemoteApplicationPublishedBridge(SpringCache eventSource) {
+      super(eventSource);
+      registerListener(eventSource);
+   }
+
+   private final void registerListener(SpringCache eventSource) {
+      ((RemoteCache<?, ?>) eventSource.getNativeCache()).addClientListener(this);
```

Comment:
I just noticed this.... I let `this` reference to escape here... I need to correct this!